### ISSUE TITLE
twisted.internet.abstract.FileDescriptor.__init__ reactor parameter is undocumented

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6793]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6793|
|Type|enhancement|
|Created|2013-10-20 14:54:41Z|
|Branch|https://github.com/twisted/twisted/tree/filedescriptor-doc-6793|

It should have API documentation.

<details><summary>Searchable metadata</summary>

```
trac-id__6793 6793
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_filedescriptor_doc_6793 branches/filedescriptor-doc-6793
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1382280881000000 1382280881000000
changetime__1382312621000000 1382312621000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [40408]) Branching to 'filedescriptor-doc-6793'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_3"></a>|@wallrj set owner to @exarkun|
|-|-|

exarkun,

This looks fine...unless you want to add an @type string too. I thought that was required by twistedchecker, but it hasn't flagged it here, so maybe you got away with it.

Anyway, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [40414]) Merge filedescriptor-doc-6793

Author: exarkun
Reviewer: rwall
Fixes: [#6793](https://github.com/twisted/twisted/issues/6793)

Add API documentation for the `reactor` parameter to the initializer
for `twisted.internet.abstract.FileDescriptor`.
